UI/pre-stream-wizard/encoder-settings-provider-facebook.cpp

```diff
@@ -0,0 +1,303 @@
+{
+	blog(LOG_INFO, "FacebookEncoderSettingsProvider sending request");
+
+	bool requestSuccess = false;
+
+	std::string urlString = url.toString().toStdString();
+	std::string reply;
+	std::string error;
+	long responseCode = 0;
+	const char *contentType = "application/json";
+	const char *postData = nullptr;
+	std::vector<std::string> extraHeaders = std::vector<std::string>();
+	int timeout = 3; // seconds
+
+	auto apiRequestBlock = [&]() {
+		requestSuccess = GetRemoteFile(urlString.c_str(), reply, error,
+					       &responseCode, contentType,
+					       postData, extraHeaders, nullptr,
+					       timeout);
+	};
+
+	ExecuteFuncSafeBlock(apiRequestBlock);
+
+	if (!requestSuccess || responseCode >= 400) {
+		handleTimeout();
+		blog(LOG_WARNING, "Server response with error: %s",
+		     error.c_str());
+	}
+
+	if (reply.empty()) {
+		handleEmpty();
+		blog(LOG_WARNING, "Server response was empty");
+	}
+
+	QByteArray jsonBytes = QByteArray::fromStdString(reply);
+	handleResponse(jsonBytes);
+}
+
+QUrlQuery FacebookEncoderSettingsProvider::inputVideoQueryFromCurrentSettings()
+{
+	// Get input settings, shorten name
+	EncoderSettingsRequest *input = currentSettings_.data();
+
+	QUrlQuery inputVideoSettingsQuery;
+	inputVideoSettingsQuery.addQueryItem("video_type", "live");
+	if (input->userSelectedResolution.isNull()) {
+		inputVideoSettingsQuery.addQueryItem(
+			"input_video_width",
+			QString::number(input->videoWidth));
+		inputVideoSettingsQuery.addQueryItem(
+			"input_video_height",
+			QString::number(input->videoHeight));
+	} else {
+		QSize wizardResolution = input->userSelectedResolution.toSize();
+		inputVideoSettingsQuery.addQueryItem(
+			"input_video_width",
+			QString::number(wizardResolution.width()));
+		inputVideoSettingsQuery.addQueryItem(
+			"input_video_height",
+			QString::number(wizardResolution.height()));
+	}
+	inputVideoSettingsQuery.addQueryItem("input_video_framerate",
+					     QString::number(input->framerate));
+	inputVideoSettingsQuery.addQueryItem(
+		"input_video_bitrate", QString::number(input->videoBitrate));
+	inputVideoSettingsQuery.addQueryItem(
+		"input_audio_channels", QString::number(input->audioChannels));
+	inputVideoSettingsQuery.addQueryItem(
+		"input_audio_samplerate",
+		QString::number(input->audioSamplerate));
+	if (input->protocol == StreamProtocol::rtmps) {
+		inputVideoSettingsQuery.addQueryItem("cap_streaming_protocols",
+						     "rtmps");
+	}
+	// Defaults in OBS
+	inputVideoSettingsQuery.addQueryItem("cap_video_codecs", "h264");
+	inputVideoSettingsQuery.addQueryItem("cap_audio_codecs", "aac");
+	return inputVideoSettingsQuery;
+}
+
+QString FacebookEncoderSettingsProvider::getOBSVersionString()
+{
+	QString versionString;
+
+#ifdef HAVE_OBSCONFIG_H
+	versionString += OBS_VERSION;
+#else
+	versionString += LIBOBS_API_MAJOR_VER + "." + LIBOBS_API_MINOR_VER +
+			 "." + LIBOBS_API_PATCH_VER;
+#endif
+
+	return versionString;
+}
```

Comment:
Instead of potentially having every provider have to duplicate this code, which is itself already present elsewhere in OBS code, I'd rather see it as a general function either in the pre-stream-wizard utils or somewhere more general, like obs-app.

Reply:
I ended up not using this, I'll just remove this.